### PR TITLE
Handle gtpu sequence number

### PR DESCRIPTION
**Pull request by orion-belt**

### CODE REVIEW

File: src/spgwu/simpleswitch/spgwu_s1u.cpp [hunk 1]
```diff
@@ -123,8 +123,14 @@ void spgwu_s1u::handle_receive(
...
     if (gtpuh->message_type == GTPU_G_PDU) {
       uint8_t gtp_flags = recv_buffer[GTPU_MESSAGE_FLAGS_POS_IN_UDP_PAYLOAD];
+      std::pair<bool, uint16_t> gtp_seqn;
       std::size_t gtp_payload_offset = GTPV1U_MSG_HEADER_MIN_SIZE;
       if (gtp_flags == 0x34) gtp_payload_offset += 4;
+      if (gtp_flags == 0x36) {
+        gtp_payload_offset += 4;
+        gtp_seqn.first  = true;
+        gtp_seqn.second = gtpuh->sequence;
+      }
```
Comment (lionelgo) [inline src/spgwu/simpleswitch/spgwu_s1u.cpp:133]: I would recommend be..toh()

**Comment by orion-belt** [comment]
Hi @lionelgo
I removed copying of gtpu seqn in DL as we may not need it.
I also not process gtp seqn (just added offfset for decoding).

Is that okay ?

**Comment by orion-belt** [comment]
Hi @lionelgo 
Will you please review changes ?
thanks

File: src/spgwu/simpleswitch/spgwu_s1u.cpp [hunk 2]
```diff
@@ -116,6 +116,9 @@ void spgwu_s1u::handle_receive(
     char* recv_buffer, const std::size_t bytes_transferred,
     const endpoint& r_endpoint) {
 #define GTPU_MESSAGE_FLAGS_POS_IN_UDP_PAYLOAD 0
+#define GTPU_MESSAGE_EXT_HEADER_FLAG 0x34
+#define GTPU_MESSAGE_SEQN_FLAG 0x36
+#define GTPU_MESSAGE_PRIME_FLAG 0x07
```
Comment (lionelgo) [inline src/spgwu/simpleswitch/spgwu_s1u.cpp:121]: These constants could be set in gtpv1u/gtpu.h
Ext header bit for example is not 0x34 but 0x4, you optimize the code, but the compiler will do it for you

Comment (orion-belt) [reply src/spgwu/simpleswitch/spgwu_s1u.cpp:121]: HI Lionel, What  0x34 should be defined ?

Comment (orion-belt) [reply src/spgwu/simpleswitch/spgwu_s1u.cpp:121]: Because it is combination of type, version and ext_hdr_present flag.

Comment (lionelgo) [reply src/spgwu/simpleswitch/spgwu_s1u.cpp:121]: You could do this:
#define GTP_VERSION_1 1
#define GTPU_MESSAGE_SN_MASK 0x2
#define GTPU_MESSAGE_EXT_HEADER_MASK 0x4
#define GTPU_MESSAGE_PT_MASK 0x10
#define GTPU_MESSAGE_VERSION_MASK 0xE
(in gtpu.h)
then 
if (
     (
       (gtp_flags & GTPU_MESSAGE_VERSION_MASK) == GTP_VERSION_1) && (gtp_flags & GTPU_MESSAGE_PT_MASK)
     ) 
    && 
     (
        (gtp_flags & GTPU_MESSAGE_EXT_HEADER_MASK) || (gtp_flags & GTPU_MESSAGE_SN_MASK)
     )
   ) {
...
}

Comment (orion-belt) [reply src/spgwu/simpleswitch/spgwu_s1u.cpp]: @lionelgo 
Looks okay now ?

Comment (lionelgo) [reply src/spgwu/simpleswitch/spgwu_s1u.cpp]: One more thing : we should also test the PN flag (forgot to mention previously)

**Review by lionelgo — APPROVED** [review]

**Pull request merged**